### DISCUSSION TITLE
resume bookmark,nothing happened!

### DISCUSSION THREAD

**Discussion by yutao815** [elsa-workflows/elsa-core#4472] [Q&A]
I create a bookmark,use WorkflowInbox to resume the bookmark,but nothing happend!
1、Create bookmark,publish a event!
![image](https://github.com/elsa-workflows/elsa-core/assets/26317013/6b55e536-3230-45b8-aee7-b812175514b0)
2、Handle the event result.
![image](https://github.com/elsa-workflows/elsa-core/assets/26317013/ff5f0351-5d27-4ff1-a20d-32778b21810a)
What I need to do.Thanks for reply.

**Comment by sfmskywalker** [comment]
Hi @yutao815,

Thank you for reporting this issue. Nothing jumps out at me at first sight, so let's figure it out as follows:

1. Please fork the repository.
2. Update the `Elsa.WorkflowServer.Web` reference application with your custom activity and logic to resume the activity.
3. Share the link to your fork with me so I can reproduce and troubleshoot the issue on my local machine.

Alternatively, if you want to just send me a solution that reproduces this issue that would work too of course.

Looking forward to your response!

**Comment by yutao815** [comment]
https://github.com/yutao815/elsa-core-fork/tree/v3/src/bundles/Elsa.WorkflowServer.Web

**Comment by sfmskywalker** [comment]
Hi @yutao815 

Thanks for providing the repro!
I see now why it's not working:

- In the `NewWorkflowEndpoint`, you're creating a new workflow on the fly that only exists during the lifetime of that endpoint.
- Therefore, when you try to resume that workflow from `ResumeWorkflowEndpoint`, it won't find that workflow, because it no longer exists.

When working with resumable workflows, they need to be made available to the runtime. Just like if you were to generate a .NET function at runtime, you need to keep it around in order to be able to invoke it again.

I hope that makes sense, but let me know if you I can be of further assistance!

**Comment by yutao815** [comment]
Thanks for reply.Could your give me a sample?

**Comment by sfmskywalker** [comment]
Sure! [Here](https://github.com/elsa-workflows/elsa-core/tree/v3/src/samples/aspnet/Elsa.Samples.AspNet.HelloWorld) you will find an example of a workflow that is declared once and registered with the system.
